From fa302400eec06acdf0d3a7fafb9137e0c9c30a97 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Wed, 5 Jun 2024 14:35:20 +0200
Subject: [PATCH 0904/1695] scpfw: fix CFG_SCPFW_MOD_* for product modules

Build directives CFG_SCPFW_MOD_PSU_OPTEE_REGULATOR and
CFG_SCPFW_MOD_STM32_PD refer to product specific modules in
SCP-firmware hence these directives must be defined in the
product CMakeLists.txt files, not in the arch/none/opteeCMakeLists.txt
file.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 8da13d1f17b5eb9822b10097077e0c8e104cbe47 scpfw: fix CFG_SCPFW_MOD_* for product modules]
Change-Id: I31a5f61bf53fbb54f982c7d58e6eeeec9fe22f66
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/385438
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 .../scmi-server/SCP-firmware/arch/none/optee/CMakeLists.txt    | 1 -
 .../SCP-firmware/product/optee-stm32mp1/fw/CMakeLists.txt      | 2 ++
 .../SCP-firmware/product/optee-stm32mp2/fw/CMakeLists.txt      | 3 +++
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/core/lib/scmi-server/SCP-firmware/arch/none/optee/CMakeLists.txt b/core/lib/scmi-server/SCP-firmware/arch/none/optee/CMakeLists.txt
index 1cb21fac8..c65672fcb 100644
--- a/core/lib/scmi-server/SCP-firmware/arch/none/optee/CMakeLists.txt
+++ b/core/lib/scmi-server/SCP-firmware/arch/none/optee/CMakeLists.txt
@@ -15,7 +15,6 @@ set (CFG_SCPFW_MOD_MSG_SMT n)
 set (CFG_SCPFW_MOD_OPTEE_CONSOLE n)
 set (CFG_SCPFW_MOD_OPTEE_CLOCK n)
 set (CFG_SCPFW_MOD_OPTEE_MBX n)
-set (CFG_SCPFW_MOD_PSU_OPTEE_REGULATOR n)
 set (CFG_SCPFW_MOD_OPTEE_RESET n)
 set (CFG_SCPFW_MOD_OPTEE_SMT n)
 set (CFG_SCPFW_MOD_OPTEE_VOLTD_REGULATOR n)
diff --git a/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp1/fw/CMakeLists.txt b/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp1/fw/CMakeLists.txt
index 8674a1eb9..a7ef941de 100644
--- a/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp1/fw/CMakeLists.txt
+++ b/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp1/fw/CMakeLists.txt
@@ -9,6 +9,8 @@
 # Create the firmware target.
 #
 
+set (CFG_SCPFW_MOD_PSU_OPTEE_REGULATOR n)
+
 add_library(${SCP_FIRMWARE_TARGET})
 
 # cmake-lint: disable=E1122
diff --git a/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp2/fw/CMakeLists.txt b/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp2/fw/CMakeLists.txt
index 8674a1eb9..6a85a8d79 100644
--- a/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp2/fw/CMakeLists.txt
+++ b/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp2/fw/CMakeLists.txt
@@ -9,6 +9,9 @@
 # Create the firmware target.
 #
 
+set (CFG_SCPFW_MOD_PSU_OPTEE_REGULATOR n)
+set (CFG_SCPFW_MOD_STM32_PD n)
+
 add_library(${SCP_FIRMWARE_TARGET})
 
 # cmake-lint: disable=E1122
-- 
2.47.2

